### PR TITLE
Refactor deployment examples

### PR DESCRIPTION
**Pull request by Alevsk**
- Adding Resources field for KES configuration
- Added deployment examples to be run with kustomize build command

Signed-off-by: Lenin Alevski <alevsk.8772@gmail.com>

### CODE REVIEW

**Comment by Alevsk** [comment]
I'm investigating the following error displayed on operator during tenant users provisioning

```
tenant-external-idp-ldap/storage-external-idp-ldap': This 'admin' API is not supported by server in 'mode-server-distributed-xl
```

when `LDAP` is enabled for a particular tenant, most likely its a incompatible version between madmin library and the minio version running by the tenant

File: examples/kustomization/base/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: minio-tenant
+resources:
+  - namespace.yaml
+  - tenant-config.yaml
+  - tenant-minio-config_deprecated.yaml
```
Comment (harshavardhana) [inline examples/kustomization/base/kustomization.yaml:7]:
```suggestion
  - tenant-minio-config_deprecated.yaml
```

why is this deprecate.yaml?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: examples/kustomization/base/kustomization.yaml (hunk 1 above)
Comment (Alevsk) [reply examples/kustomization/base/kustomization.yaml:7]: we are passing the tenant credentials via the `tenant-config.yaml`, configuration values there will have more precedence than `credsSecret` or tenant env variables, also credentials will not appear if someone do `docker inspect` of a minio pod

File: examples/kustomization/base/tenant-minio-config_deprecated.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
```
Comment (dvaldivia) [inline examples/kustomization/base/tenant-minio-config_deprecated.yaml:1]: I think this file should be called `tenant-minio-creds-secret_deprecated.yaml`

**Review by dvaldivia — CHANGES_REQUESTED** [review]
Please update the readme with a set of examples on how to use the examples via `kubectl -k`

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**